Project-1/Peihong_Xie_Project_1_50375781.md

```diff
@@ -0,0 +1,249 @@
+Tip #4: The work we do is challenging; that should be assumed. You are smart enough to be here; that should also be assumed. We have neither time nor space for shaming, but all of time and space for praising. Be cognizant of how your messages might be received, and err on the side of caution. It is hard to surmise intent from text alone. For my part, I treat text only communications the way modern musicals are written: Little subtext; emotions on the sleeve. 
+```
+
+Note: The standard interpretation of the logical symbols - "∨", "∧", "→", "¬", "∀", "∃" - is assumed throughout. 
+
+[1] Provide the truth tables for each of the following propositional logic formulas. State whether each is a tautology, a contradiction, or contingent:
+  ```(a) (¬A→B)∨((A∧¬C)→B) 
+  (b) (A→B)∧(A→¬B)
+  (c) (A→(B∨C))∨(C→¬A) 
+  (d) ((A→B)∧C)∨(A∧D) 
+```
+
+Answer: 
+
+(a) Tautology
+| A | B	| C | ¬A→B | A∧¬C | (A∧¬C)→B | (¬A→B)∨((A∧¬C)→B) |
+| ---  | --- | --- | --- | --- | --- | --- |
+| 1 | 1 | 1 | 1 | 0 | 1 | 1 |
+| 1 | 1 | 0 | 1 | 1 | 1 | 1 |
+| 1 | 0 | 1 | 1 | 0 | 1 | 1 |
+| 0 | 1 | 1 | 1 | 0 | 1 | 1 |
+| 1 | 0 | 0 | 1 | 1 | 0 | 1 |
+| 0 | 1 | 0 | 1 | 0 | 1 | 1 |
+| 0 | 0 | 1 | 0 | 0 | 1 | 1 |
+| 0 | 0 | 0 | 0 | 0 | 1 | 1 |
+
+(b) Contingent
+| A | B	| A→B | A→¬B | (A→B)∧(A→¬B) | 
+| ---  | --- | --- | --- | --- |
+| 1 | 1 | 1 | 0 | 0 | 
+| 1 | 0 | 0 | 1 | 0 |
+| 0 | 1 | 1 | 1 | 1 |
+| 0 | 0 | 1 | 1 | 1 | 
+
+(c) Tautology
+| A | B	| C | B∨C | A→(B∨C) | C→¬A | (A→(B∨C))∨(C→¬A)) |
+| ---  | --- | --- | --- | --- | --- | --- |
+| 1 | 1 | 1 | 1 | 1 | 0 | 1 |
+| 1 | 1 | 0 | 1 | 1 | 1 | 1 |
+| 1 | 0 | 1 | 1 | 1 | 0 | 1 |
+| 0 | 1 | 1 | 1 | 1 | 1 | 1 |
+| 1 | 0 | 0 | 0 | 0 | 1 | 1 |
+| 0 | 1 | 0 | 1 | 1 | 1 | 1 |
+| 0 | 0 | 1 | 1 | 1 | 1 | 1 |
+| 0 | 0 | 0 | 0 | 1 | 1 | 1 |
+
+(d) Contingent
+| A | B	| C | D | A→B | (A→B)∧C | A∧D | ((A→B)∧C)∨(A∧D)
+| ---  | --- | --- | --- | --- | --- | --- |--- |
+| 1  | 1 | 1 | 1 | 1 | 1 | 1 | 1 |
+| 1  | 1 | 1 | 0 | 1 | 1 | 0 | 1 |
+| 1  | 1 | 0 | 1 | 1 | 0 | 1 | 1 |
+| 1  | 0 | 1 | 1 | 0 | 0 | 1 | 1 |
+| 0  | 1 | 1 | 1 | 1 | 1 | 0 | 1 |
+| 1  | 1 | 0 | 0 | 1 | 0 | 0 | 0 |
+| 1  | 0 | 1 | 0 | 0 | 0 | 0 | 0 |
+| 0  | 0 | 1 | 1 | 1 | 1 | 0 | 1 |
+| 0  | 1 | 1 | 0 | 1 | 1 | 0 | 1 |
+| 0  | 1 | 0 | 1 | 1 | 0 | 0 | 0 |
+| 1  | 0 | 0 | 1 | 0 | 0 | 1 | 1 |
+| 0  | 0 | 0 | 1 | 1 | 0 | 0 | 0 |
+| 0  | 0 | 1 | 0 | 1 | 1 | 0 | 1 |
+| 0  | 1 | 0 | 0 | 1 | 0 | 0 | 0 |
+| 1  | 0 | 0 | 0 | 0 | 0 | 0 | 0 |
+| 0  | 0 | 0 | 0 | 1 | 0 | 0 | 0 |
+
+[2] A _literal_ is an atomic formula or the negation of an atomic formula. We say a formula is in _conjunctive normal form_ (CNF) if it is the conjunction of the disjunction of literals. Find propositional logic formulas in CNF equivalent to each of the following:
+  ```(a) (A→B)→C
+  (b) (A→(B∨C))∨(C→¬A)
+  (c) (¬A∧¬B∧C)∨(¬A∧¬C)∨(B∧C)∨A 
+```
+Answer:
+
+(a) CNF: (A∨C)∧(¬B∨C)
+
+(b) CNF: B∨¬A
+
+(c) CNF: B∨¬B (the CNF of a tautology is any formula of the form "p∨¬p")
```

Comment:
Could you point me to where you drew this conclusion?

Reply:
Regarding (c), if you run it in a truth table generator like https://web.stanford.edu/class/cs103/tools/truth-table-tool/, you will find it is a tautology because every line of its truth-table is True. 

Regarding (b), I calculate it manually: 
(A→(B∨C))∨(C→¬A)
(1) ((~A)∨(B∨C))∨((~C)∨(~A))
(2) delete a (~A) because (~A)∨(~A) is equivalent to (~A): (B∨C)∨(~C)∨(~A)
(3) delete C∨(~C) because this is a tautology: B∨(~A)